acvm_backend_barretenberg/src/pedersen.rs

```diff
@@ -0,0 +1,155 @@
+use common::acvm::FieldElement;
+
+use super::Barretenberg;
+
+impl Barretenberg {
+    pub(crate) fn compress_native(
+        &mut self,
+        left: &FieldElement,
+        right: &FieldElement,
+    ) -> FieldElement {
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "native")] {
+                let result_bytes = barretenberg_sys::pedersen::compress_native(
+                    left.to_be_bytes().as_slice().try_into().unwrap(),
+                    right.to_be_bytes().as_slice().try_into().unwrap(),
+                );
+            } else {
+                use wasmer::Value;
+
+                let lhs_ptr = self.allocate(&left.to_be_bytes()); // 0..32
```

Comment:
I don't think these comments are correct anymore since allocate will not place 32 bytes, starting from pointer address 0. `lhs_ptr` and `rhs_ptr` are allocated somewhere.

With that in mind, you could do `let result_ptr = Value::I32(0);` and then `let result_bytes = self.slice_memory(0, 32);`